packages/app/src/cli/models/app/loader.ts

```diff
@@ -574,19 +580,25 @@ async function logMetadataForLoadedApp(
   })
 }
 
-export function isValidAppConfigurationFile(config: string) {
-  const validFileRegex = /^shopify\.app(\.[-\w]+)?\.toml$/g
-  return validFileRegex.test(config)
-}
+export const appConfigurationFileNameRegex = /^shopify\.app(\.[-\w]+)?\.toml$/
 
 export function getAppConfigurationFileName(config?: string) {
-  if (config) {
-    if (isValidAppConfigurationFile(config)) {
-      return config
-    }
+  if (!config) {
+    return configurationFileNames.app
+  }
 
-    return `shopify.app.${config}.toml`
+  if (appConfigurationFileNameRegex.test(config)) {
+    return config
+  } else {
+    return `shopify.app.${slugify(config)}.toml`
   }
+}
 
-  return configurationFileNames.app
+export function getAppConfigurationShorthand(path: string) {
+  const match = basename(path).match(appConfigurationFileNameRegex)
+  if (match) {
+    const result = match[1]?.slice(1)
+    if (result === '.') return undefined
+    return result
+  }
```

Comment:
I think the result will never be `.` because `"shopify.app.toml".match(/^shopify\.app(\.[-\w]+)?\.toml$/)[1]?.slice(1)` returns undefined, so we can simplify this.

```suggestion
  return match?.[1]?.slice(1)
```

Reply:
Oh right makes sense!